### PR TITLE
Add Logging 5.5 and new profile for Logging 5.7 and other fixes

### PR DESCRIPTION
**Pull request by IshwarKanse**
As part of this PR:

* Add Logging 5.5 periodic job.
* Add profiles for Logging 5.7
* Logging 5.7 uses the openshift-tests-private release-4.13 branch.
* Use Logging CPaaS index.
* Fix job messages in reporter config. Previously the job message in slack was showing as failed for any job state. Now the job message will show that the job is executed and the status can be seen from the logs.
* Tweak the cron entries for the jobs.
* Add external grafana loki credential and function to the openshift-extended-logging-test step to export the creds for use with ginkgo test cases.

### CODE REVIEW

**Comment by IshwarKanse** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-openshift-logging-5.7-gcp-ipi-ovn-ipsec-p1-f3-openshift-logging-5-7

File: ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11-periodics.yaml
Comment (QiaolingTang) [reply ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11-periodics.yaml:16402]: release-4.13?

**Comment by IshwarKanse** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-openshift-logging-5.7-aws-ipi-proxy-sdn-workers-rhel8-p1-f7-openshift-logging-5-7

**Comment by IshwarKanse** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-openshift-logging-5.7-aws-ipi-proxy-cco-manual-sts-p1-f7-openshift-logging-5-7

File: ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11-periodics.yaml
Comment (IshwarKanse) [reply ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11-periodics.yaml:16402]: Added release-4.14 profile for Logging 5.7. For the ginkgo branch, checking with test platform team as to why the generated job is using base_ref as release-4.11 even when we have specified the tag release-4.13 for openshift-tests-private image in the ci-operator config for the job.

Comment (IshwarKanse) [reply ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11-periodics.yaml:16402]: @anpingli @QiaolingTang I checked with Kui Wang for the Ginkgo branch config. The config we are using in this PR is correct. 

1 The cluster build is configured by [this config](https://github.com/openshift/release/pull/37208/files#diff-d08055a4adccc9bc31e46fc7bc9927cba29cad8051597b8a385bb84e0ddc70d2R40-R45), it decides we use 4.11 to create cluster.
2 The golang step take [image](https://github.com/openshift/release/pull/37208/files#diff-d08055a4adccc9bc31e46fc7bc9927cba29cad8051597b8a385bb84e0ddc70d2R18-R21) to create pod to start test. it is 4.13 branch image, so we take 4.13 test code to start cases.
3, [Our concern](https://github.com/openshift/release/pull/37208/files#diff-4ff7234a64bf60c6570775ca7f3d769feb9bde181bf8e5883f58b8ac8f3f5ba1R16402) is decided by [here](https://github.com/openshift/release/pull/37208/files#diff-d08055a4adccc9bc31e46fc7bc9927cba29cad8051597b8a385bb84e0ddc70d2R88-R91). it is correct for 4.11, we can not change it. even if we change it, it will be corrected by prow. it does not impact our test code. it just impact the image pipeline:root, pipeline:src, and pipeline:bin which are not used by golang test step.

I had ran [this](https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/37208/rehearse-37208-periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-openshift-logging-5.7-aws-ipi-proxy-cco-manual-sts-p1-f7-openshift-logging-5-7/1635554621160165376) rehearsal which seems to have ran fine.

There is one more commit I need to add to this PR for the external hosted Grafana Loki instance. I'll add that today.

Comment (anpingli) [reply ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11-periodics.yaml:16402]: Got it, thanks

**Comment by IshwarKanse** [comment]
/pj-rehearse ci/rehearse/periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-openshift-logging-5.7-gcp-ipi-ovn-ipsec-p1-f3-openshift-logging-5-7

**Comment by IshwarKanse** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-openshift-logging-5.7-gcp-ipi-ovn-ipsec-p1-f7-openshift-logging-5-7

**Comment by IshwarKanse** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-openshift-logging-5.7-gcp-ipi-ovn-ipsec-p1-f7-openshift-logging-5-7

File: ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11-periodics.yaml
Comment (IshwarKanse) [reply ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11-periodics.yaml:16402]: @QiaolingTang @anpingli Added the external grafana loki credential. Let me know if there is anything else needed as part of the PR and if we are good to merge it.

**Comment by IshwarKanse** [comment]
/pj-rehearse ci/rehearse/periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-openshift-logging-5.7-gcp-ipi-ovn-ipsec-p1-f3-openshift-logging-5-7

**Comment by IshwarKanse** [comment]
/pj-rehearse ci/rehearse/periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-openshift-logging-5.7-aws-ipi-proxy-sdn-workers-rhel8-p1-f7-openshift-logging-5-7

**Comment by IshwarKanse** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-arm64-nightly-openshift-logging-5.7-aws-ipi-sdn-p1-f7-openshift-logging-5-7

**Comment by IshwarKanse** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-arm64-nightly-openshift-logging-5.7-aws-ipi-sdn-p1-f7-openshift-logging-5-7

**Comment by IshwarKanse** [comment]
@QiaolingTang @anpingli Added the external grafana loki credential. Let me know if there is anything else needed as part of the PR and if we are good to merge it.

**Comment by IshwarKanse** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-arm64-nightly-openshift-logging-5.7-aws-ipi-sdn-p1-f7-openshift-logging-5-7

**Comment by IshwarKanse** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-openshift-logging-5.7-gcp-ipi-ovn-ipsec-p1-f7-openshift-logging-5-7

**Comment by IshwarKanse** [comment]
```
started: (0/10/144) "[sig-openshift-logging] Logging NonPreRelease Test forward logs to external Grafana Loki log store CPaasrunOnly-Author:ikanse-Medium-48923-Vector Forward logs to Grafana Loki using HTTPS and existing loki.tenantKey kubernetes.namespace_name[Serial]"
passed: (1m35s) 2023-03-16T07:28:40 "[sig-openshift-logging] Logging NonPreRelease Test forward logs to external Grafana Loki log store CPaasrunOnly-Author:ikanse-Medium-48923-Vector Forward logs to Grafana Loki using HTTPS and existing loki.tenantKey kubernetes.namespace_name[Serial]"
started: (0/11/144) "[sig-openshift-logging] Logging NonPreRelease Test forward logs to external Grafana Loki log store CPaasrunOnly-Author:ikanse-Medium-48489-Vector Forward logs to Grafana Loki using HTTPS [Serial]"
passed: (1m31s) 2023-03-16T07:30:11 "[sig-openshift-logging] Logging NonPreRelease Test forward logs to external Grafana Loki log store CPaasrunOnly-Author:ikanse-Medium-48489-Vector Forward logs to Grafana Loki using HTTPS [Serial]"
```
The external grafana loki cases are working now with the cred added to the ginkgo test step.

**Comment by kuiwang02** [comment]
I am OK from goalng's view.

but you need to request qe-productivity team to approve it.

**Comment by IshwarKanse** [comment]
@liangxia Need your review and approval for this PR.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-openshift-logging-5.5.yaml
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-openshift-logging-5.5.yaml:54]: Shall we just add the test into existing profiles `openshift-openshift-tests-private-release-4.xxx__amd64-nightly.yaml` as we do for hypershift/rosa/osd/...

Comment (IshwarKanse) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-openshift-logging-5.5.yaml:54]: We have tied the logging versions to the corresponding openshift-tests-private branches, where Logging 5.7 is tied to release-4.13 and Logging 5.6 is tied to release-4.12. To ensure consistency, we have added new configuration files. Even if we need to add tests to existing profiles, creating new config files for specifying a different openshift-tests-private branch for a particular Logging version running on different OCP versions is necessary. With separate configurations, maintenance becomes easier for us.

Comment (IshwarKanse) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-openshift-logging-5.5.yaml:54]: Can we override the tests-private and tests-private-ui images when adding the tests? For example for running Logging 5.7 tests, we would need tag 4.13 when added to the profiles using different OCP versions as you mentioned. openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml, openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml and openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml,

  tests-private:
    name: tests-private
    namespace: ci
    tag: "4.13"
  tests-private-ui:
    name: tests-private-ui
    namespace: ci
    tag: "4.13"

Comment (IshwarKanse) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-openshift-logging-5.5.yaml:54]: I do not see any other issue that prevents us from adding the Logging tests to existing profile configs. Its just the tests-private branch image that we need to override.

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-openshift-logging-5.5.yaml:54]: In `openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml`, tests-private and tests-private-ui is already 4.13, refer to https://github.com/openshift/release/blob/master/ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml#L22-L37
,  so I think there is no need to override

Comment (IshwarKanse) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-openshift-logging-5.5.yaml:54]: I mean in openshift-openshift-tests-private-release-4.11__amd64-nightly.yaml, openshift-openshift-tests-private-release-4.12__amd64-nightly.yaml, is it possible to override the tests-private tag?

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-openshift-logging-5.5.yaml:54]: If you check the `zz_generated_metadata` part, you can see each file/test is related to proper branch. E.g, for 4.13,
```
zz_generated_metadata:
  branch: release-4.13
  org: openshift
  repo: openshift-tests-private
  variant: amd64-nightly
```
May i know the reason why do you want to use 4.13 branch to do tests on 4.11/4.12 ?

Comment (IshwarKanse) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-openshift-logging-5.5.yaml:54]: Our Logging version tests code is dependent on the openshift-tests-private branch. The tests for running Logging 5.7 is in release-4.13 branch of the openshift-tests-private repo. Thus if we want to run Logging 5.7 on any of the OCP versions(4.11,4.12,4.13,4.14), we would need to use the release-4.13 branch of the openshif-tests-private repo. Unfortunately the decision for tying Logging versions to tests-private release branches was done when we initially started switching to using ginkgo for our test runs. cc @anpingli @QiaolingTang

Comment (IshwarKanse) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-openshift-logging-5.5.yaml:54]: BTW these tests are for full functionality testing for Logging and will be used for Logging releases.

Comment (anpingli) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-openshift-logging-5.5.yaml:54]: @liangxia  
Logging 5.5 cases are in branch release-4.11, Logging 5.6 cases are in branch release-4.12, Logging 5.7 cases are in branch release-4.13.
So if we want to test Logging 5.5 on OCP 4.13.  Our team must use release-4.11.

Comment (anpingli) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-openshift-logging-5.5.yaml:54]: These profiles are used to run logging function cases, which can not be ran in parallel. and all logging case will take about 5 hours.  I don't think these cases can be executed using existing profiles.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly-openshift-logging-5.7.yaml
Comment (liangxia) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly-openshift-logging-5.7.yaml:26]: Are we using upgrade images during e2e testing ?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly-openshift-logging-5.7.yaml
Comment (IshwarKanse) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly-openshift-logging-5.7.yaml]: Rebased and removed the mult arch payload and upgrade images. The profile config we referenced had the arch and images. openshift-openshift-tests-private-release-4.13__amd64-nightly.yaml

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13__amd64-nightly-openshift-logging-5.7.yaml]: Because we have single arch to multi arch migration tests in 4.13 for OCP

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-openshift-logging-5.5.yaml
Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-openshift-logging-5.5.yaml:54]: Need to use 4.13 images to test 4.11 sounds like a design issue.
Anyway, I'm unholding the PR for now. Leave it to your feature team to decide.

**Comment by IshwarKanse** [comment]
@QiaolingTang Needs your LGTM on this.

**Review by danilo-gemoli — APPROVED** [review]

**Pull request closed without merging**
